server/server.go

```diff
@@ -567,3 +570,30 @@ func (s *Server) handleListClients(user *auth.User, w http.ResponseWriter, r *ht
 		TokenData: resp.RefreshTokens,
 	})
 }
+
+func (s *Server) handleOpenShiftTokenDeletion(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		sendResponse(w, http.StatusMethodNotAllowed, apiError{"Invalid method: only POST is allowed"})
+		return
+	}
+
+	// Delete the OpenShift OAuthAccessToken.
+	path := "/apis/oauth.openshift.io/v1/oauthaccesstokens/" + user.Token
+	url := proxy.SingleJoiningSlash(s.K8sProxyConfig.Endpoint.String(), path)
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		sendResponse(w, http.StatusInternalServerError, apiError{fmt.Sprintf("Failed to create token DELETE request: %v", err)})
```

Comment:
What would cause a failure to create the request?

Reply:
a URL that can't be parsed or an error reading the request body (although we don't have a body here)

Reply:
Source is here if you're curious:

https://golang.org/src/net/http/request.go

Reply:
Right! Was more thinking, we have all the bits to create the URL here, not too likely for it to fail, nothing is user-defined.  I'm good with keeping this check in place, either way, just working through the details.

Reply:
We should definitely be checking `err`. `NewRequest` is a black box to us. We shouldn't assume it will always work even if we're confident the URL is valid.